plugins/pluginmanagers/lazy.nix

```diff
@@ -0,0 +1,187 @@
+        name = "${lib.getName p.pkg}";
+        path = p.pkg;
+      };
+    processDependencies =
+      if plugin ? dependencies && plugin.dependencies != null
+      then builtins.concatMap processPlugin plugin.dependencies
+      else [];
+  in
+    [(mkEntryFromDrv plugin)] ++ processDependencies;
+
+  processedPlugins = builtins.concatLists (builtins.map processPlugin lazyPlugins);
+  lazyPath = pkgs.linkFarm "lazy-plugins" processedPlugins;
+in {
+  options = {
+    plugins.lazy = {
+      enable = mkEnableOption "lazy.nvim";
+
+      plugins = with types; let
+        pluginType =
+          either
+          package
+          (submodule {
+            options = {
+              dir = helpers.mkNullOrOption str "A directory pointing to a local plugin";
+
+              pkg = mkOption {
+                type = package;
+                description = "Vim plugin to install";
+              };
+
+              name = helpers.mkNullOrOption str "Name of the plugin to install";
+
+              dev = helpers.defaultNullOpts.mkBool false "When true, a local plugin directory will be used instead. See config.dev";
+
+              lazy = helpers.defaultNullOpts.mkBool true "When true, the plugin will only be loaded when needed. Lazy-loaded plugins are automatically loaded when their Lua modules are required, or when one of the lazy-loading handlers triggers";
+
+              enabled = helpers.defaultNullOpts.mkBool true "When false then this plugin will not be included in the spec";
+
+              cond =
+                helpers.mkNullOrOption (helpers.nixvimTypes.maybeRaw bool)
+                "When false, or if the function returns false, then this plugin will not be loaded. Useful to disable some plugins in vscode, or firenvim for example.";
+
+              dependencies =
+                helpers.mkNullOrOption (helpers.nixvimTypes.eitherRecursive str listOfPlugins)
+                "Plugin dependencies";
+
+              init =
+                helpers.mkNullOrOption (helpers.nixvimTypes.maybeRaw bool)
+                "init functions are always executed during startup";
+
+              config =
+                helpers.mkNullOrOption (helpers.nixvimTypes.maybeRaw bool)
+                "config is executed when the plugin loads. The default implementation will automatically run require(MAIN).setup(opts). Lazy uses several heuristics to determine the plugin's MAIN module automatically based on the plugin's name. See also opts. To use the default implementation without opts set config to true.";
+
+              main = helpers.mkNullOrOption str "You can specify the main module to use for config() and opts(), in case it can not be determined automatically. See config()";
+
+              submodules = helpers.defaultNullOpts.mkBool true "When false, git submodules will not be fetched. Defaults to true";
+
+              event = with helpers.nixvimTypes;
+                helpers.mkNullOrOption (maybeRaw (either str (listOf str)))
```

Comment:
The type seems pretty complex:
```
string? or string[] or fun(self:LazyPlugin, event:string[]):string[] or {event:string[]|string, pattern?:string[]|string}
```

That seems to translate to `let item = oneOf [str (listOf str) rawLua]; in either item (listOf item)`

Reply:
somehow the nix version reads easier than the description... yet doesn't quite "compile"

Reply:
What  do you mean it doesn't "compile" ?

Reply:
Though I think I mistyped the option, it looks like more of a `oneOf [str (listOf str) rawLua (submodule  {event = ...; pattern= ...;})];`, but we could skip the submodule, and simply use `oneOf [str (listOf str) rawLua]`.

You can choose what you prefer